crates/client/src/client.rs

```diff
@@ -335,9 +335,46 @@ impl Client {
             .expect("get_balance task failed")?
     }
 
+    /// Deploys the program and wraps the boilerplate code related to the deployment.
+    #[throws]
+    pub async fn deploy_program(&self, client: &Client, program_keypair: &Keypair, name: &str) {
+        debug!("reading program data");
+
+        let reader = Reader::new();
+        let mut program_data = reader
+            .program_data(name)
+            .await
+            .expect("reading program data failed");
+        let program_data_len = program_data.len();
+
+        debug!("calculation minimum rent expemtion balance");
+
+        let system_program = self.anchor_client.program(System::id());
+        let rpc_client = system_program.rpc();
+        let min_balance_for_rent_exemption = 2 * task::spawn_blocking(move || {
+            rpc_client.get_minimum_balance_for_rent_exemption(program_data_len)
+        })
+        .await
+        .expect("crate program account task failed")?;
+
+        debug!("airdropping the minimum balance required to deploy the program");
+
+        client
+            .airdrop(client.payer().pubkey(), min_balance_for_rent_exemption)
+            .await
+            .expect("airdropping for deployment failed");
+
+        debug!("deploying program");
+
+        client
+            .deploy(program_keypair.clone(), mem::take(&mut program_data))
```

Comment:
It is confusing to have two methods called deploy. The user does not know the difference and which one to call. So I would merge them. By that, we also avoid code duplication (both methods must get the RPC client, find out the minimum rent exempt amount, ...). 

Maybe I would add an example call to the documentation and explain the `name` argument.

Reply:
I didn't want to create one huge function with over ~120 lines of code. I believe that keeping these two split makes the code more readable and maintanable at the cost of a few boilerplate lines.
The old `deploy` function is not `pub`lic, so I don't think the user has to (or even can) care which function to call.